numba/unicode.py

```diff
@@ -869,6 +872,13 @@ def unicode_strip_types_check(chars):
         raise TypingError('The arg must be a UnicodeType or None')
 
 
+def args_types_check(arg):
+    if not (arg is None or isinstance(arg, (types.Omitted,
+                                            types.Optional,
+                                            types.Integer))):
```

Comment:
This should also accept `types.NoneType` for cases like:
```python
from numba import njit

@njit
def foo(x, y, none_ty=None):
    return x.count(y, none_ty, 1)

a = "ascii"
b = "a"

print(foo.py_func(a, b))
print(foo(a, b))
```
where start is a `NoneType` opposed to the literal `None`. A test should be added to ensure this is correctly implemented.